components/net/fetch/request.rs

```diff
@@ -553,15 +552,15 @@ fn http_network_or_cache_fetch(request: Rc<RefCell<Request>>,
 
     // Step 1
     let http_request = if request_has_no_window &&
-        request.borrow().redirect_mode != RedirectMode::Follow {
+        request.redirect_mode != RedirectMode::Follow {
         request.clone()
     } else {
-        Rc::new(RefCell::new(request.borrow().clone()))
+        request.clone()
```

Comment:
guh, I thought I got that one before, too! I've reworked that whole bit, the spec requires the if statement which I added a todo about